homeassistant/components/media_player/apple_tv.py

```diff
@@ -0,0 +1,250 @@
+https://home-assistant.io/components/media_player.apple_tv/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK,
+    SUPPORT_STOP, SUPPORT_PLAY, MediaPlayerDevice, PLATFORM_SCHEMA,
+    MEDIA_TYPE_MUSIC, MEDIA_TYPE_VIDEO, MEDIA_TYPE_TVSHOW)
+from homeassistant.components.http import HomeAssistantView, KEY_AUTHENTICATED
+from homeassistant.const import (
+    STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_STANDBY, CONF_HOST,
+    CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['pyatv==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ARTWORK_URL = "/api/apple_tv/{0}/artwork.png?token={1}"
+
+CONF_HSGID = 'hsgid'
+
+DEFAULT_NAME = 'Apple TV'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_HSGID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_entities,
+                         discovery_info=None):
+    """Setup the Apple TV platform."""
+    entity = AppleTvDevice(
+        hass,
+        name=config.get(CONF_NAME),
+        host=config.get(CONF_HOST),
+        hsgid=config.get(CONF_HSGID))
+
+    yield from async_add_entities([entity], update_before_add=True)
+
+
+class AppleTvDevice(MediaPlayerDevice):
+    """Representation of an Apple TV device."""
+
+    def __init__(self, hass, name, host, hsgid):
+        """Initialize the Apple TV device."""
+        self.hass = hass
+        self._name = name
+        self._playing = None
+        self.artwork = None
+        hass.http.register_view(ArtworkImageView(self))
+
+        import pyatv
+        details = pyatv.AppleTVDevice(name, host, hsgid)
+        self._atv = pyatv.connect_to_apple_tv(details, hass.loop)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self._playing is not None:
+            from pyatv import const
+            state = self._playing.play_state
+            if state == const.PLAY_STATE_NO_MEDIA:
+                return STATE_IDLE
+            elif state == const.PLAY_STATE_PLAYING or \
+                    state == const.PLAY_STATE_LOADING:
+                return STATE_PLAYING
+            elif state == const.PLAY_STATE_PAUSED or \
+                    state == const.PLAY_STATE_FAST_FORWARD or \
+                    state == const.PLAY_STATE_FAST_BACKWARD:
+                # Catch fast forward/backward here so "play" is default action
+                return STATE_PAUSED
+            else:
+                return STATE_STANDBY  # Bad or unknown state?
+
+    @property
+    def access_token(self):
```

Comment:
You shouldn't have to implement this yourself - it is implemented in the base class.